core/src/toga/icons.py

```diff
@@ -1,10 +1,16 @@
 from __future__ import annotations
 
 from pathlib import Path
+from typing import TYPE_CHECKING
 
 import toga
 from toga.platform import get_platform_factory
 
+if TYPE_CHECKING:
+    from typing import TypeAlias
```

Comment:
As a note, we can avoid this by depending on the `typing_extensions` package; it backports all the new stuff for typing.

Reply:
Ah - good to know. I'll add that in.

Reply:
Of course, it all gets screwed up by Py3.12 saying that the preferred syntax is now `type X = Y`...

Reply:
Also - we can't avoid doing the import in the TYPE_CHECKING block unless we make typing-extensions a base-level dependency (rather than a dev dependency). I'm hesitant to add a runtime requirement that by definition does nothing at runtime.

Reply:
> Also - we can't avoid doing the import in the TYPE_CHECKING block unless we make typing-extensions a base-level dependency (rather than a dev dependency). I'm hesitant to add a runtime requirement that by definition does nothing at runtime.

hmm....it'll probably make more sense to discuss this in #2252....I have thoughts but they don't need to burden this PR :)